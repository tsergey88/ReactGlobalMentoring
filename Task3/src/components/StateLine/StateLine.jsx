import React from "react";

export class StateLine extends React.Component {
  render() {
    return (
      <div className="state-line">
        <div className="state-line-left">7 movies found</div>
        <div className="state-line-right">
          <div>Sort by</div>
          <div>release date</div>
          <div className="active">rating</div>
        </div>
      </div>
    );
  }
}
