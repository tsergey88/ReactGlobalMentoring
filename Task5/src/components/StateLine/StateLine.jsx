// @flow
import React from "react";

type Props = {
  left: string,
  right: JSX.Element
};

export const StateLine = ({ left, right }: Props) => (
  <div className="state-line">
    <div className="state-line-left">{left}</div>
    <div className="state-line-right">{right}</div>
  </div>
);
