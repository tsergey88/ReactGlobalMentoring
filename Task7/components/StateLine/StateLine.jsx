// @flow
import React from "react";

import styles from "./style.scss";

type Props = {
  left: string,
  right: JSX.Element
};

export const StateLine = ({ left, right }: Props) => (
  <div className={styles.state_line}>
    <div>{left}</div>
    <div className={styles.state_line_right}>{right}</div>
  </div>
);
