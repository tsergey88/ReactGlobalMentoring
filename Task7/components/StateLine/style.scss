.state_line {
  display: flex;
  justify-content: space-between;
  padding: 5px 10em;
  font-weight: 900;
  color: black;
  background-color: #dddddd;
}
.state_line_right {
  display: flex;

  div {
    margin: 0 10px;
  }

  .active {
    color: #d24979;
  }
}