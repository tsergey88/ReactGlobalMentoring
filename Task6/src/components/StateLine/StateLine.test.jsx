import React from "react";
import { StateLine } from "./StateLine";
import { shallow } from "enzyme";

const props = {
  left: "Left",
  right: "Right"
};

describe("StateLine  component", () => {
  const component = shallow(<StateLine {...props} />);

  it("should be render correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("should contain correct div", () => {
    expect(component.find("div")).toHaveLength(3);
  });
});
