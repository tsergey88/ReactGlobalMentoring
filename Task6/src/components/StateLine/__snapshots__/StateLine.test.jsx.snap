// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StateLine  component should be render correctly 1`] = `
<div
  className="state-line"
>
  <div
    className="state-line-left"
  >
    Left
  </div>
  <div
    className="state-line-right"
  >
    Right
  </div>
</div>
`;
